### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by MrCreosote**
... or alternately, sthu dependabot

This was much more of a pain than I expected when I started on it.

Also see https://github.com/kbase/relation_engine/issues/156

- [n/a] I updated the README.md docs to reflect this change.

For changes to the codebase:

- [n/a] I have written tests to cover this change.
- [x] This is not a breaking API change OR
- [ ] This is a breaking API change and I have incremented the API version and added a summary to CHANGELOG.md.

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -1,10 +1,9 @@
-mypy>=0.630
-bandit==1.5.1
+mypy>=0.971
+bandit==1.7.4
+# mccabe 0.7.0 is not compatible with the flake8 4.0.1
 mccabe==0.6.1
 flake8==4.0.1
-grequests==0.3.0
-coverage==5.2.1
-typed-ast>=1.4.0
```
Comment (MrCreosote) [inline dev-requirements.txt:7]: Not needed in python 3.8+

File: dev-requirements.txt [hunk 2]
```diff
@@ -1,10 +1,9 @@
...
-black==22.3.0
-pytest==6.2.5
-jinja2==3.0.3
```
Comment (MrCreosote) [inline dev-requirements.txt:10]: Already in `requirements.txt`

File: relation_engine_server/utils/json_validation.py [hunk 3]
```diff
@@ -12,12 +12,10 @@
 """
 from jsonschema import validators, Draft7Validator, FormatChecker, RefResolver
 
-from jsonschema.compat import (
```
Comment (MrCreosote) [inline relation_engine_server/utils/json_validation.py:15]: `jsonschema.compat` was an internal-only API and was removed in 4.X (without a mention in the release notes either...)

Comment (n1mus) [reply relation_engine_server/utils/json_validation.py:15]: Hmm I tried to do this before and ran into errors

Comment (n1mus) [reply relation_engine_server/utils/json_validation.py:15]: Okay you got the functions from somewhere else. Nice.

File: relation_engine_server/utils/json_validation.py [hunk 4]
```diff
@@ -165,7 +163,8 @@ def resolve_remote(self, uri):
             result = requests.get(uri).json()
         else:
             # Otherwise, pass off to urllib and assume utf-8
-            with urlopen(uri) as url:
+            # TODO https://github.com/kbase/relation_engine/issues/156
```
Comment (n1mus) [inline relation_engine_server/utils/json_validation.py:166]: Seems like this is for the remote specs loading that we don't even use? Maybe? If so, YAGNI for now

File: scripts/run_tests.sh [hunk 5]
```diff
@@ -20,6 +20,6 @@ python -m spec.validate
 python -m relation_engine_server.utils.wait_for api
 # run importer/, relation_engine_server/, and spec/ tests
 coverage run --branch -m pytest
-# RE client tests - seemss like these tests were already run in the line above, why run them again?
+# RE client tests - seems like these tests were already run in the line above, why run them again?
```
Comment (n1mus) [inline scripts/run_tests.sh:23]: Yay

**Review by n1mus — APPROVED** [review]
LGTM. Nice!

File: relation_engine_server/utils/json_validation.py (hunk 4 above)
Comment (MrCreosote) [reply relation_engine_server/utils/json_validation.py:166]: Can we confirm we don't use remote specs and we're not going to? If so we could just remove this class (and maybe more stuff)

**Pull request merged**